### PR TITLE
fix: User/UserClient and Fleet/Fleet client classes

### PR DESCRIPTION
**Pull request by robercano**
- Fixes the implementation of UserClient and FleetClient by having proper User and Fleet base classes
- Fleet has been renamed to EarnProtocolFleet
- Fixed E2E tests and tested Earn Protocol against the endpoint to make sure the changes are working
- Minor fixes
- Several changes are due to import order fixing by VSCode plugin

### CODE REVIEW

**Review by piotrwitek — CHANGES_REQUESTED** [review]
looks great I just just spotted few improvements to consider

File: sdk/sdk-e2e/tests/earnProtocol.test.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const DAI = Token.createFrom({
 
 /** TEST CONFIG */
 const config = {
-  SDKApiUrl: 'https://h6bwee4lvb.execute-api.us-east-1.amazonaws.com/api/sdk',
+  SDKApiUrl: 'https://72dytt1e93.execute-api.us-east-1.amazonaws.com/api/sdk',
```
[the remark this answers is not in the document]
Comment (robercano) [reply sdk/sdk-e2e/tests/earnProtocol.test.ts]: Yes, although we need to talk about how to properly do the E2E tests first (like having a proper DevNet for them, and an SDK endpoint so they can be run. For now E2E tests are skipped altogether during `cicheck`

File: sdk/sdk-client/src/implementation/EarnProtocolManagerClient.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ export class EarnProtocolManagerClient extends IRPCClient implements IEarnProtoc
   }
 
   /** @see IEarnProtocolManagerClient */
-  public getFleet(params: { address: IAddress }): Maybe<IFleetClient> {
-    return new FleetClient({
+  public getFleet(params: { address: IAddress }): Maybe<IEarnProtocolFleetClient> {
+    return new EarnProtocolFleetClient({
       ...params,
```
[the remark this answers is not in the document]
Comment (robercano) [reply sdk/sdk-client/src/implementation/EarnProtocolManagerClient.ts:25]: So the benefit of doing that is that any new parameters are automatically handled. The con is that you may effectively be passing more parameters. Not sure what is best to have

[the remark this answers is not in the document]
Comment (robercano) [reply sdk/sdk-client/src/implementation/EarnProtocolManagerClient.ts]: Agreed to destructure it in cases where we are merging different types of parameters like this one, and not to destructure it when `params` can be passed untouched

File: sdk/sdk-e2e/tests/earnProtocol.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (robercano) [reply sdk/sdk-e2e/tests/earnProtocol.test.ts]: Agreed to have a discussion offline to implement this at a later time

**Review by piotrwitek — APPROVED** [review]

**Pull request merged**